app/forms/schools/coaches/update_form.rb

```diff
@@ -26,6 +26,8 @@ class UpdateForm < Reform::Form
       property :affiliation, virtual: true
 
       def save
+        # Important to call this service before update
+        Schools::ArchiveCoachService.new(faculty, exited).execute
```

Comment:
Keep this inside the transaction. 
https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html

Reply:
Also you can revert all changes in this file;